app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
I think @dLobatog meant that the host that you try to provision always has architecture assigned otherwise you can't select OS. On the other hand host can be created multiple ways which does not always enforce architecture being present, therefore +1 for keeping architecture optional.